usefulUtilities/hiFiCalendar/clientScripts/meetingRoomSetup_app.js

```diff
@@ -0,0 +1,161 @@
+                Entities.callEntityServerMethod(TOKEN_SERVER_ID, "initializeRooms", [
+                    token, 
+                    refreshToken, 
+                    expireTime, 
+                    timezone,
+                    timezoneOffset, 
+                    JSON.stringify(roomConfig)
+                ]);
+                break;
+        }
+    }
+
+
+    // This function is the listener for messages on the message mixer related to this app.
+    var messageHandler = function(channel, message, senderUUID, localOnly) {
+        if (channel !== CHANNEL) {
+            return;
+        } else {
+            try {
+                message = JSON.parse(message);
+            } catch (e) {
+                console.log(e, "Could not parse message");
+                return;
+            }
+
+            if (message.TOKEN_SERVER_ID) {
+                TOKEN_SERVER_ID = message.TOKEN_SERVER_ID;
+            }
+
+            if (message.type === "TOKEN EXPIRED") {
+                Window.alert("The authorization token for your Google Calendar could not be refreshed.\n" + 
+                "Please open your calendar app and reauthorize to continue displaying calendar schedules.");
+                roomConfig = message.roomConfig;
+                roomConfigInfo = message.roomConfigInfo;
+            } else if (message.type === "STATUS UPDATE") {
+                if (message.tokenStatus) {
+                    Window.announcement("Access tokens are valid. No action required");
+                    roomConfig = message.roomConfig;
+                    roomConfigInfo = message.roomConfigInfo;
+                } else if (!message.tokenStatus && !message.roomConfigured) {
+                    roomConfigInfo = message.roomConfigInfo;
+                    Window.announcement("You have not set up any room schedules for this domain.");
+                }
+            } else if (message.type === "ERROR") {
+                // Commenting this out for right now because this was initially flooding the interface during testing
+                // Will bring back after battle testing for QA
+                
+                // Window.alert("There was an error, here's what we know:\n" + 
+                // message.entityName + "\nError: " + 
+                // message.errorMessage + "\nHappened during: " + 
+                // message.attemptedAction);
+                return;
+            } else if (message.type === "REFRESH SUCCESS") {
+                roomConfigInfo = message.roomConfigInfo;
+                Window.announcement("Token server successfully refreshed " + message.count + " times since preload.");
+            }
+            ui.sendToHtml({
+                type: "AVAILABLE ROOMS",
+                roomConfig: roomConfigInfo
+            });
+        }
+    };
+
+
+    // This function keeps the app running even if you change domains.
+    var DOMAIN_DELAY = 100;
+    function onDomainChange(domain){
```

Comment:
is this necessary?

Reply:
I think so, it's been helpful for testing at least so you know what is happening when switching domains.  Any additional thoughts?  Could go either way, I think it's more something we would see in testing than production for sure.

Reply:
After looking at it again from Huffman's additional comment, you are right. I don't think this is necessary at all.